src/blueprint/blueprint.rs

```diff
@@ -18,6 +20,26 @@ use crate::lambda::{Expression, Lambda};
 pub struct Blueprint {
   pub definitions: Vec<Definition>,
   pub schema: SchemaDefinition,
+  pub server: Server,
+}
+
+#[derive(Clone, Debug, Default, Setters)]
+pub struct Server {
+  pub allowed_headers: Option<HashSet<String>>,
+  pub base_url: Option<String>,
+  pub enable_apollo_tracing: Option<bool>,
+  pub enable_cache_control_header: Option<bool>,
+  pub enable_graphiql: Option<String>,
+  pub enable_http_cache: Option<bool>,
+  pub enable_introspection: Option<bool>,
+  pub enable_query_validation: Option<bool>,
+  pub enable_response_validation: Option<bool>,
+  pub global_response_timeout: Option<i64>,
+  pub port: Option<u16>,
+  pub upstream: Upstream,
+  pub vars: BTreeMap<String, String>,
+  pub batch: Option<crate::config::Batch>,
+  pub response_headers: HeaderMap,
 }
```

Comment:
We can drop `Option` The `Server` in blueprint should contain the final values. Which means basically no option types, HeaderMap instead of KeyValues, Urls instead of String etc. It should be safe to use `blueprint::Server`.